src/opset/Gemm_13.c

```diff
@@ -0,0 +1,187 @@
+/*
+ *  CONNX, C implementation of Open Neural Network Exchange Runtime
+ *  Copyright (C) 2019-2022 TSN Lab, Inc.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+#include <connx/accel.h>
+#include <connx/connx.h>
+
+#define GEMM_DEBUG 0
+
+__attribute__((unused)) static void show_elements_tensor(char* name, int32_t rows, int32_t cols, float32_t* d,
+                                                         bool transposed) {
+    fprintf(stderr, "%s %s Transposed ( %d, %d)\n", name, transposed ? "" : "Not", rows, cols);
+    int32_t num = transposed ? rows : cols;
+    float32_t(*m)[num] = (float32_t(*)[num])d;
+    for (int32_t r = 0; r < rows; r++) {
+        for (int32_t c = 0; c < cols; c++) {
+            float32_t val = transposed ? m[c][r] : m[r][c];
+            fprintf(stderr, "%f ", val);
+        }
+        fprintf(stderr, "\n");
+    }
+    fprintf(stderr, "\n");
+}
+
+// clang-format off
+int Gemm_{{op_version}}(connx_Graph* graph, __attribute__((unused)) uint32_t output_count, uint32_t* outputs,
+                         // clang-format on
+                         __attribute__((unused)) uint32_t input_count, uint32_t* inputs,
+                         __attribute__((unused)) uint32_t attribute_count, __attribute__((unused)) void** attributes) {
+
+    connx_Tensor* A = connx_Graph_get(graph, inputs[0]);
+    connx_Tensor* B = connx_Graph_get(graph, inputs[1]);
+    assert(A != NULL && B != NULL);
+
+    float32_t alpha = *(float32_t*)attributes[0];
+    float32_t beta = *(float32_t*)attributes[1];
+    bool transA = *(int32_t*)attributes[2] ? true : false;
```

Comment:
For opset<=6, `attributes[2]` is broadcast, And `attributes[3]`, `[4]` is the transposes

FYI: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#Gemm-6